### PR TITLE
Add CloudantValidator

### PR DESCRIPTION
**Pull request by jameswinkler**

### CODE REVIEW

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            string uri = fingerprint.Uri;
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            // At this point account and password must be either both full or both empty.  Only check one
+            if (string.IsNullOrWhiteSpace(account))
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpResponseMessage response = client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead).GetAwaiter().GetResult();
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs:86]: >HttpResponseMessage [](start = 20, length = 19)

dispose

File: Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            else
+            {
+                HttpClientHandler handler = new HttpClientHandler();
+                handler.Credentials = new NetworkCredential(account, password);
+                using (HttpClient client = new HttpClient(handler)
+                {
+                    BaseAddress = new Uri(uri),
+                })
+                {
+                    HttpResponseMessage response = client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead).GetAwaiter().GetResult();
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs:104]: >HttpResponseMessage [](start = 20, length = 19)

dispose

Comment (eddynaka) [inline Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs:104]: >client.GetAsync [](start = 51, length = 15)

add a trycatch

File: Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs (hunk 1 above)
Comment (eddynaka) [inline Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs:86]: >client.GetAsync [](start = 51, length = 15)

add try catch

File: Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs (hunk 2 above)
Comment (jameswinkler) [reply Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs]: Done

---
In reply to: [586923935](https://github.com/microsoft/sarif-pattern-matcher/pull/238#discussion_r586923935) [](ancestors = 586923935)

Comment (jameswinkler) [reply Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs]: Done

---
In reply to: [586924090](https://github.com/microsoft/sarif-pattern-matcher/pull/238#discussion_r586924090) [](ancestors = 586924090)

File: Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs (hunk 1 above)
Comment (jameswinkler) [reply Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs]: Done

---
In reply to: [586923764](https://github.com/microsoft/sarif-pattern-matcher/pull/238#discussion_r586923764) [](ancestors = 586923764)

Comment (jameswinkler) [reply Src/Plugins/Security/SEC101_048.CloudantConnectionStringValidator.cs]: Done

---
In reply to: [586924131](https://github.com/microsoft/sarif-pattern-matcher/pull/238#discussion_r586924131) [](ancestors = 586924131)

**Review by eddynaka — APPROVED** [review]

**Pull request merged**